HW06.Rmd

```diff
@@ -1,92 +1,122 @@
+  bind_rows()
+
+WT_vector <- rep("WT", length.out = nrow(WT_tables))
+WT_tables$Group <- WT_vector
+
+H3K27M_vector <- rep("H3K27M", length.out = nrow(H3K27M_tables))
+H3K27M_tables$Group <- H3K27M_vector
+
+tables <- rbind(WT_tables, H3K27M_tables)
+
+tables_TPM <- tables %>%
+  select(Name, TPM, Group)
+
+GBM.transcripts <- tables_TPM %>%
+  rename(gene_id = Name)
+
+#you only need the 
+
+```
+
+Now, install the packages commented below (if needed), and then use this code to map the transcript IDs to gene symbols. To use this code, you need a dataframe called GBM.transcripts that has the first column "gene_id" that contains the transcript ids (e.g. ENST00000456328.2) and the remaining columns contain the TPM data. So long as the first column contains the "gene_id" column as mentioned above, this *should* run. 
+
+```{r map gene.ids to gene.symbols using ensembldb}
+
+#install.packages("BiocManager")
+#BiocManager::install("ensembldb")
+#BiocManager::install("EnsDb.Hsapiens.v75")
+library(ensembldb)
+library(EnsDb.Hsapiens.v75)
+
+ens.GBM.transcripts <- GBM.transcripts %>% 
+  mutate(gene_id = gsub(pattern = "\\..*", "", .$gene_id))
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+
+```
+
+
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+genes_of_interest <- c("IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "PTEN", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2", "WNT5A")
+
+
+GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+
+#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
+
+# this didn't work, probably in part because I didn't organize the files as intended for the mapping code. But here is where I stop.
+for(genes_of_interest in ens.mapped_GBM$gene_symbol) {
```

Comment:
I might be wrong or totally not understanding what you're trying to do, but I think genes_of_interest takes on the value of each item in ens.mapped_GBM$gene_symbol as the for loop iterates through ens.mapped_GBM$gene_symbol.

I think you could set up a for loop that could work with the dataframe that you have. One way that might work would be to iterate through genes_of_interest, filter for each gene in genes_of_interest in your dataframe, and perform a t-test.